modules/core/src/main/scala/com/azavea/stac4s/StacCatalog.scala

```diff
@@ -1,33 +1,82 @@
 package com.azavea.stac4s
 
 import cats.Eq
+import cats.implicits._
 import io.circe._
+import io.circe.syntax._
 
 final case class StacCatalog(
     stacVersion: String,
     stacExtensions: List[String],
     id: String,
     title: Option[String],
     description: String,
-    links: List[StacLink]
+    links: List[StacLink],
+    extensionFields: JsonObject
 )
 
 object StacCatalog {
 
+  val catalogFields = Set(
```

Comment:
This definition of "fields" seems like the riskiest part of this strategy. Do you think there is a way to write tests that would catch incorrectly adding or removing fields?

Reply:
It's not hard to ensure that no fields from the case class are also keys in the `extensionFields` json object -- we can use [`LabelledGeneric`](https://svejcar.dev/posts/2019/10/22/extracting-case-class-field-names-with-shapeless/) from shapeless to extract the field names, and I think we can just map `_type` => `type` in the names and end up with something that mostly works. I'm trying to think of whether this is a terrible idea for some reason I haven't thought of yet. I have end-of-day brain though so I'll think about it again tomorrow morning.

I actually ran into this issue while working on this PR -- I'd forgotten the `filter` in collection and had about a half hour of trouble figuring out what was wrong with my collection json round trips.

Reply:
I like the idea of using `LabelledGeneric` a bunch - seems like something that will be useful to keep the strings in sync.

Reply:
ade5ae9 -- I tried out the generic strategy in the linked blog post but had trouble with the implicits, so we have a less generic generics strategy here

Reply:
@jisantuc what was decided btw?

Reply:
They're used in the linked commit